src/platform/ESP32/NetworkCommissioningDriver.h

```diff
@@ -0,0 +1,129 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <esp_wifi.h>
+#include <platform/NetworkCommissioning.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace NetworkCommissioning {
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
+class ESPScanResponseIterator : public Iterator<WiFiScanResponse>
+{
+public:
+    ESPScanResponseIterator(size_t size, wifi_ap_record_t * scanResults) : mSize(size), mpscanResults(scanResults) {}
```

Comment:
should input scan results be `const` as iterator should not change them?

Could you add a comment about lifetimes (i.e. iterator does not make a copy, so requires scan results to stay valid for the duration of the iteration.

Reply:
Done